Frontend/src/Components/Dashboard/Dashboard.js

```diff
@@ -126,7 +136,14 @@ export default function Dashboard(props) {
         .then((res) => {
           //console.time("update react");
 
-          setState({ data: res.response });
+          // when the car/data generator is offline, res.response is going to be null,
+          //  so we need to set variables to a default value that will prevent a TypeError
+          //  when we try reading the properties of an undefined variable
+          if (res.response === null) {
+            setState({ data: null }); // state.data should not exist in this case
+          } else {
+            setState({ data: res.response });
+          }

```

Comment:
```suggestion
          setState({ data: res.response });
```
Since `res.response` will be null when we want to set `state.data` to null, we can remove the condition.